xiblint/__main__.py

```diff
@@ -31,7 +32,11 @@ def main():
     )
     parser.add_argument("-v", "--version", action="version",
                         version=__version__)
-    parser.parse_args()
+    parser.add_argument("--reporter", choices=("raw", "json"),
+                        default="raw",
+                        help="custom reporter to use (optional)")
+    args = parser.parse_args()
+    reporter = args.reporter
```

Comment:
After you assign this variable, you're never changing it, so perhaps directly using `args.reporter` will make it clearer that it's coming from the args.